apache/rocketmq/v2/admin.proto

```diff
@@ -38,6 +43,105 @@ message ChangeLogLevelRequest {
 
 message ChangeLogLevelResponse { string remark = 1; }
 
+message DescribeTopicStatusRequest {
+  optional string instance_id = 1;
+  Resource topic = 2;
+}
+
+message DescribeTopicStatusResponse {
+  Status status = 1;
+  MessageType topic_message_type = 2;
+  string description = 3;
+  google.protobuf.Timestamp create_timestamp = 4;
+  map<string, string> tags = 5;
+}
+
+message DescribeTopicSubscriptionRequest {
+  optional string instance_id = 1;
+  Resource topic = 2;
+}
+
+message DescribeTopicSubscriptionResponse {
+  message SubscriptionInfo {
+    Resource group = 1;
+    FilterExpression expression = 2;
+    bool online = 3;
+    google.protobuf.Timestamp last_update_timestamp = 4;
+  }
+  Status status = 1;
+  repeated SubscriptionInfo subscriptionInfo = 2;
+}
+
+message DescribeConsumerAccumulateRequest {
+  optional string instance_id = 1;
+  Resource group = 2;
+  repeated Resource topics = 3;
+}
+
+message DescribeConsumerAccumulateResponse {
+  Status status = 1;
+}
+
+message DescribeConsumerStatusRequest {
+  optional string instance_id = 1;
+  Resource group = 2;
+}
+
+message DescribeConsumerStatusResponse {
+  Status status = 1;
+  int64 inflight_messages = 2;
+  int64 ready_messages = 3;
+  int64 accumulation = 4;
+  google.protobuf.Duration deliver_delay_time = 5;
+}
+
+message ResetConsumeOffsetRequest {
+  optional string instance_id = 1;
+  Resource group = 2;
+  Resource topic = 3;
+  google.protobuf.Timestamp reset_timestamp = 4;
+}
+
+message ResetConsumeOffsetResponse {
+  Status status = 1;
+}
+
+message QueryMessageByIdRequest {
```

Comment:
Will it be feasible to combine QueryMessageByIdRequest and QueryMessageByKeyRequest by having QueryMessageRequest and conducting actual queries based on what is provided in the optional fields?